src/init.cpp

```diff
@@ -887,12 +869,11 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     const CChainParams& chainparams = Params();
 
     // Set this early so that experimental features are correctly enabled/disabled
-    if (!InitExperimentalMode()) {
-        return false;
+    auto res = InitExperimentalMode();
+    if (res != "success") {
+        return InitError(res);
```

Comment:
Instead of returning a string and then matching on its contents (which is brittle), return a `boost::optional<std::string>` and then make this something like:
```cpp
auto err = InitExperimentalMode();
if (err) {
    return InitError(err.get());
}
```

Reply:
A lot better, thank you. Implemented at https://github.com/zcash/zcash/pull/4293/commits/17fb20e61f9862d8a5f947a0f4c7e69b194428a7